### PR TITLE
tests - ceph-ansible backports

### PR DESCRIPTION
**Pull request by yuriw**
This is backport of PRs:

https://github.com/ceph/ceph/pull/16826
https://github.com/ceph/ceph/pull/17096
https://github.com/ceph/ceph/pull/17650
https://github.com/ceph/ceph/pull/17757

Signed-off-by: Yuri Weinstein yweinste@redhat.com

### CODE REVIEW

**Comment by yuriw** [comment]
@vasukulkarni how do we test this?

**Comment by vasukulkarni** [comment]
@yuriw thanks for the backports, it needs ceph-ansible suite, ceph-deploy suite and upgrade(with filter ceph-deploy), All for luminous branch.

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml [hunk 1]
```diff
@@ -1,6 +1,14 @@
+<<<<<<< HEAD
```
Comment (vasukulkarni) [inline qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml]: Need to fix this Conflict

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
 os_type: centos
 os_version: "7.3"
 machine_type: vps
+=======
+machine_type: ovh
+openstack:
+- volumes: # attached to each instance
+    count: 3 
+    size: 10 # GB
```
Comment (cbodley) [inline qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml]: discard the HEAD, and keep this section

Comment (yuriw) [reply qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml]: done

**Comment by yuriw** [comment]
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-09-27_02:48:10-ceph-ansible-luminous-distro-basic-vps/
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-09-27_02:48:46-ceph-deploy-luminous-distro-basic-vps/
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-09-27_02:50:16-upgrade:jewel-x:ceph-deploy-luminous-distro-basic-vps/

**Comment by vasukulkarni** [comment]
@yuriw  looks great
a) minor ceph-deploy issues during cleanup , the lock files held on /var/lib/mon during tar
b) ceph-ansible looks good too, the xenial SSL_INIT issue(during purge) i think we can check with @djgalloway to see what workaround we can use in venv for purge-cluster, we already have system-site-enabled and install works fine
c) upgrade worked fine!

**Comment by vasukulkarni** [comment]
we can also run rgw (with --filter s3a , 2 tests) to see how it goes, but there is a tracker related to it. but we can merge this for now.

**Pull request merged**